src/passes/stringify-walker-impl.h

```diff
@@ -0,0 +1,120 @@
+#include "stringify-walker.h"
+
+#ifndef wasm_passes_stringify_walker_impl_h
+#define wasm_passes_stringify_walker_impl_h
+
+namespace wasm {
+
+template<typename SubType>
+inline void StringifyWalker<SubType>::doWalkModule(Module* module) {
+  ModuleUtils::iterDefinedFunctions(*module, [&](Function* func) {
+    this->walkFunction(func);
+  });
+}
+
+template<typename SubType>
+inline void StringifyWalker<SubType>::doWalkFunction(Function* func) {
+  walk(func->body);
+  addUniqueSymbol();
+}
+
+template<typename SubType>
+inline void StringifyWalker<SubType>::walk(Expression* curr) {
+  Super::walk(curr);
+  do {
+    addUniqueSymbol();
+    dequeueControlFlow();
+  } while (!controlFlowQueue.empty());
+}
+
+template<typename SubType>
+inline void StringifyWalker<SubType>::scan(SubType* self, Expression** currp) {
+  Expression* curr = *currp;
+  if (Properties::isControlFlowStructure(curr)) {
+    self->controlFlowQueue.push(currp);
+    self->pushTask(doVisitExpression, currp);
+    if (auto* iff = curr->dynCast<If>()) {
+      Super::scan(self, &iff->condition);
```

Comment:
This logic is somewhat similar to the special handling of If here:

https://github.com/WebAssembly/binaryen/blob/84316e8fc3448932c4d62d1e749047aaacf02ef2/src/ir/iteration.h#L143-L155

I wonder if we can write this code using ValueIterator..? Not sure.

Reply:
done! thanks for the tip!